datasources/src/main/java/org/opensearch/sql/datasources/utils/XContentParserUtils.java

```diff
@@ -48,6 +50,7 @@ public static DataSourceMetadata toDataSourceMetadata(XContentParser parser) thr
     List<String> allowedRoles = new ArrayList<>();
     Map<String, String> properties = new HashMap<>();
     String resultIndex = null;
+    DataSourceStatus status = null;
```

Comment:
what if user upgrade from 2.11 domain which datasource does not include status field? status should default to true?

Reply:
Yes, it will be Active. Even though it is null here, DatasourceMetadata builder will take care of handling null and make it as active.